File: templates/template.java
import static java.lang.Integer.parseInt;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * <h1>Day ${day} - ${title}</h1>
 * ${url}
 * <p>
 *  Start: ${fetch_time} <br>
 * Finish: TODO
 */
public class Day${day_padded} {
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {
        // Parse input
        // TODO

        // Solve
        if (args.length < 1) {
            System.err.println("Please specify the part to solve.");
            System.exit(1);
        }
        int part = parseInt(args[0]);
        switch (part) {
            case 1 -> System.out.println(part1());
            case 2 -> System.out.println(part2());
            default -> {
                System.err.println("Part must be 1 or 2.");
                System.exit(1);
            }
        }
    }

    //=============== PART 1 ===============//
    static long part1() {
        // TODO
        throw new UnsupportedOperationException("TODO");
    }

    //=============== PART 2 ===============//
    static long part2() {
        // TODO
        throw new UnsupportedOperationException("TODO");
    }

}

File: templates/template.rs
//! Day ${day} - ${title}
//! ${url}
//!  Start: ${fetch_time}
//! Finish: TODO

use std::{env::args, io::stdin};

fn main() {
    // Parse input
    for line in stdin().lines().map(|s| s.unwrap().trim().to_string()) {
        // ...
    }

    // Solve
    let answer = match args().nth(1).and_then(|s| i32::from_str_radix(&s, 10).ok()) {
        Some(1) => part1(),
        Some(2) => part2(),
        _ => lib::print_usage_and_exit!()
    };
    println!("{answer}")
}

//=============== PART 1 ===============//
fn part1() -> i64 {
    todo!()
}

//=============== PART 2 ===============//
fn part2() -> i64 {
    todo!()
}

